
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>FastViromeExplorer &#8212; FastViromeExplorer  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="fastviromeexplorer">
<span id="fastviromeexplorer"></span><h1>FastViromeExplorer<a class="headerlink" href="#fastviromeexplorer" title="Permalink to this headline">¶</a></h1>
<p>Indentify the viruses/phages and their abundance in the viral metagenomics data. FastViromeExplorer is freely available at: <a href="https://code.vt.edu/saima5/FastViromeExplorer">https://code.vt.edu/saima5/FastViromeExplorer</a></p>
</div>
<div class="section" id="installation">
<span id="installation"></span><h1>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h1>
<p>FastViromeExplorer requires JAVA (JDK) 1.8 or later, Samtools 1.4 or later, and Kallisto 0.43.0 or later installed in the user's machine.</p>
<div class="section" id="download-fastviromeexplorer">
<span id="download-fastviromeexplorer"></span><h2>Download FastViromeExplorer<a class="headerlink" href="#download-fastviromeexplorer" title="Permalink to this headline">¶</a></h2>
<p>You can download FastViromeExplorer directly from github (Github link: <a href="https://code.vt.edu/saima5/FastViromeExplorer">https://code.vt.edu/saima5/FastViromeExplorer</a>) and extract it. You can also download it using the following command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>git clone git@code.vt.edu:saima5/FastViromeExplorer.git
</pre></div>
</div>
<p>From now on, we will refer the FastViromeExplorer directory in the user's local machine as project directory. The project directory will contain 5 folders src, bin, test, tools-linux, and tools-mac. It will also contain two text file ncbi-viruses-list and imgvr-viruses-list.txt.</p>
</div>
<div class="section" id="install-java">
<span id="install-java"></span><h2>Install Java<a class="headerlink" href="#install-java" title="Permalink to this headline">¶</a></h2>
<p>If Java is not already installed, you need to install Java (JDK) 1.8 or later from the following link: <a href="http://www.oracle.com/technetwork/java/javase/downloads/index.html">http://www.oracle.com/technetwork/java/javase/downloads/index.html</a>. From this link, download the appropriate jdk installation file (for linux or macOS), and then install Java by double-clicking the downloaded installation file.
After installing Java, you can check it by running the following command in terminal:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>java -version
</pre></div>
</div>
<p>It should show you something similar to the following output:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>java version <span class="s2">&quot;1.8.0_144&quot;</span>
Java<span class="o">(</span>TM<span class="o">)</span> SE Runtime Environment <span class="o">(</span>build <span class="m">1</span>.8.0_144-b01<span class="o">)</span>
Java HotSpot<span class="o">(</span>TM<span class="o">)</span> <span class="m">64</span>-Bit Server VM <span class="o">(</span>build <span class="m">25</span>.144-b01, mixed mode<span class="o">)</span>
</pre></div>
</div>
</div>
<div class="section" id="install-kallisto-and-samtools">
<span id="install-kallisto-and-samtools"></span><h2>Install Kallisto and Samtools<a class="headerlink" href="#install-kallisto-and-samtools" title="Permalink to this headline">¶</a></h2>
<p>If Kallisto or Samtools is not installed, you can install it from the executables distributed with FastViromeExplorer.
In terminal, go into the project directory. Then go into the <code class="docutils literal"><span class="pre">tools-linux</span></code> folder if you are using a linux machine or go into the <code class="docutils literal"><span class="pre">tools-mac</span></code> folder if you are using macOS. Copy the kallisto and samtools executables from this directory to the /usr/local/bin directory.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="nb">cd</span> /path-to-FastViromeExplorer/tools-linux
sudo cp kallisto /usr/local/bin/
sudo cp samtools /usr/local/bin/
</pre></div>
</div>
<p>Or</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="nb">cd</span> /path-to-FastViromeExplorer/tools-mac
sudo cp kallisto /usr/local/bin/
sudo cp samtools /usr/local/bin/
</pre></div>
</div>
<p>After installing Kallisto and Samtools, you can check it by running the following commands to check their version.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>kallisto version
samtools
</pre></div>
</div>
<p>It will show you outputs like this in terminal:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>kallisto, version <span class="m">0</span>.43.1
</pre></div>
</div>
<div class="highlight-bash"><div class="highlight"><pre><span></span>Program: samtools <span class="o">(</span>Tools <span class="k">for</span> alignments in the SAM format<span class="o">)</span>
Version: <span class="m">1</span>.5 <span class="o">(</span>using htslib <span class="m">1</span>.5<span class="o">)</span>
...
</pre></div>
</div>
</div>
<div class="section" id="install-fastviromeexplorer">
<span id="install-fastviromeexplorer"></span><h2>Install FastViromeExplorer<a class="headerlink" href="#install-fastviromeexplorer" title="Permalink to this headline">¶</a></h2>
<p>In terminal, go into the project directory, which should contain <code class="docutils literal"><span class="pre">src</span></code> and <code class="docutils literal"><span class="pre">bin</span></code> folders. From the project directory, run the following command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>javac -d bin src/*.java
</pre></div>
</div>
<p>After installing FastViromeExplorer, you can check it by running the following command from the same project directory:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>java -cp bin FastViromeExplorer
</pre></div>
</div>
<p>It will print the usage and parameter list of FastViromeExplorer in terminal.</p>
</div>
</div>
<div class="section" id="run-fastviromeexplorer-using-test-data">
<span id="run-fastviromeexplorer-using-test-data"></span><h1>Run FastViromeExplorer using test data<a class="headerlink" href="#run-fastviromeexplorer-using-test-data" title="Permalink to this headline">¶</a></h1>
<p>From the project directory, run the following commands:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>mkdir test-output
java -cp bin FastViromeExplorer -1 test/reads_1.fq -2 test/reads_2.fq -i test/testset-kallisto-index.idx -o test-output
</pre></div>
</div>
<p>The test input files are given in the <code class="docutils literal"><span class="pre">test</span></code> folder. Here, the input files are:</p>
<ol>
<li><i>reads_1.fq</i> and <i>reads_2.fq</i> : paired-end reads in fastq format</li>
<li><i>testset-kallisto-index.idx</i> : kallisto index file generated for a small set of NCBI RefSeq viruses</li>
</ol>
The output files will be generated in the `test-output` directory. The output files are:
<ol>
<li><i>FastViromeExplorer-reads-mapped-sorted.sam</i> : aligned/mapped reads in sam format</li>
<li><i>FastViromeExplorer-final-sorted-abundance.tsv</i> : virus abundance result in tab-delimited format</li>
</ol>
In a similar manner, we can run FastViromeExplorer for single-end reads without specifying the "-2" parameter. An example of running FastViromeExplorer for single-end reads:<div class="highlight-bash"><div class="highlight"><pre><span></span>mkdir test-output
java -cp bin FastViromeExplorer -1 test/reads_1.fq -i test/testset-kallisto-index.idx -o test-output
</pre></div>
</div>
</div>
<div class="section" id="run-fastviromeexplorer-using-ncbi-refseq-database">
<span id="run-fastviromeexplorer-using-ncbi-refseq-database"></span><h1>Run FastViromeExplorer using NCBI RefSeq database<a class="headerlink" href="#run-fastviromeexplorer-using-ncbi-refseq-database" title="Permalink to this headline">¶</a></h1>
<p>Some pre-computed kallisto index files are given in the following link: <a href="http://bench.cs.vt.edu/FastViromeExplorer/">http://bench.cs.vt.edu/FastViromeExplorer/</a>.
Download the kallisto index file for NCBI RefSeq database &quot;ncbi-virus-kallisto-index-k31.idx&quot; and save it. From the project directory, run the following command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>java -cp bin FastViromeExplorer -1 <span class="nv">$read1File</span> -2 <span class="nv">$read2File</span> -i /path-to-index-file/ncbi-virus-kallisto-index-k31.idx -o <span class="nv">$outputDirectory</span>
</pre></div>
</div>
</div>
<div class="section" id="run-fastviromeexplorer-using-img-vr-database">
<span id="run-fastviromeexplorer-using-img-vr-database"></span><h1>Run FastViromeExplorer using IMG/VR database<a class="headerlink" href="#run-fastviromeexplorer-using-img-vr-database" title="Permalink to this headline">¶</a></h1>
<p>Download the kallisto index file for IMG/VR database &quot;imgvr-virus-kallisto-index-k31.idx&quot; from <a href="http://bench.cs.vt.edu/FastViromeExplorer/">http://bench.cs.vt.edu/FastViromeExplorer/</a> and save it. From the project directory, run the following command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>java -cp bin FastViromeExplorer -1 <span class="nv">$read1File</span> -2 <span class="nv">$read2File</span> -i /path-to-index-file/imgvr-virus-kallisto-index-k31.idx -l imgvr-viruses-list.txt -o <span class="nv">$outputDirectory</span>
</pre></div>
</div>
<p>For running FastViromeExplorer using IMG/VR database, we need to specify the kallisto index file and the list of viruses in the database along with their genome length, which is given in the file &quot;imgvr-viruses-list.txt&quot;.</p>
<p>After running FastViromeExplorer using IMG/VR as reference database, the abundance output will be saved in FastViromeExplorer-final-sorted-abundance.tsv. The 1st column of this file contains the ids of the mVCs. An example id is: 7000000498.a:SRS017521_Baylor_scaffold_5189. Here, the 1st part of the id (before .a:) represents a genome id from IMG/VR database and the 2nd part (after .a:) represents the scaffold id from IMG/VR database. To retrieve the annotation of a genome or of a scaffold, you need to go to the IMG/VR webpage (<a href="https://img.jgi.doe.gov/cgi-bin/vr/main.cgi">https://img.jgi.doe.gov/cgi-bin/vr/main.cgi</a>). You can find annotation of a genome by searching for the genome id in the &quot;Quick Genome Search&quot; search box. Similarly, you can find annotation of a scaffold by searching for the scaffold id in the &quot;Scaffold Search&quot; section of IMG/VR (<a href="https://img.jgi.doe.gov/cgi-bin/vr/main.cgi?section=ScaffoldSearch">https://img.jgi.doe.gov/cgi-bin/vr/main.cgi?section=ScaffoldSearch</a>)</p>
</div>
<div class="section" id="run-fastviromeexplorer-using-global-ocean-virome-gov-database">
<span id="run-fastviromeexplorer-using-global-ocean-virome-gov-database"></span><h1>Run FastViromeExplorer using Global Ocean Virome (GOV) database<a class="headerlink" href="#run-fastviromeexplorer-using-global-ocean-virome-gov-database" title="Permalink to this headline">¶</a></h1>
<p>Download the kallisto index file for GOV database &quot;GOV_viral_contigs_EPI_MES.idx&quot; and the list of viruses in the database &quot;GOV_viral_contigs_EPI_MES_list.txt&quot; from <a href="http://bench.cs.vt.edu/FastViromeExplorer/">http://bench.cs.vt.edu/FastViromeExplorer/</a> and save those files. From the project directory, run the following command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>java -cp bin FastViromeExplorer -1 <span class="nv">$read1File</span> -2 <span class="nv">$read2File</span> -i /path-to-index-file/GOV_viral_contigs_EPI_MES.idx -l /path-to-virus-list/GOV_viral_contigs_EPI_MES_list.txt -o <span class="nv">$outputDirectory</span>
</pre></div>
</div>
</div>
<div class="section" id="run-fastviromeexplorer-using-custom-database">
<span id="run-fastviromeexplorer-using-custom-database"></span><h1>Run FastViromeExplorer using custom database<a class="headerlink" href="#run-fastviromeexplorer-using-custom-database" title="Permalink to this headline">¶</a></h1>
<p>You can run FastViromeExplorer using any custom database.</p>
<p>To run FastViromeExplorer using a custom database, you need to prepare two files:</p>
<ol>
<li>A fasta file containing the reference genomes</li>
<li>A text file containing the list of reference genomes along with the genome length</li>
</ol><p>The text file should have four columns seperated by tab. Those four columns are virus-id, virus-name, virus-taxonomy, and genome-length. The 1st and last column must have values. For 2nd and 3rd columns, you can put &quot;N/A&quot; if those values are not available.
Two list of viruses are already given with the source of FastViromeExplorer, the default virus list (ncbi-viruses-list.txt) containg all NCBI RefSeq viruses and the IMG/VR virus list (imgvr-viruses-list.txt). You need to create a similar list for your custom database.</p>
<p>An example of the virus list from &quot;ncbi-viruses-list.txt&quot;:</p>
<p>NC_033618.1    Pea leaf distortion betasatellite clone N36-54, complete sequence    Unclassified;Unclassified;Unclassified;Unclassified;Unclassified;Unclassified;Pea leaf distortion betasatellite    1347</p>
<p>NC_028989.1    Pepper yellow leaf curl Thailand virus isolate KON-KG5 segment DNA-A, complete sequence    Unclassified;Unclassified;Unclassified;Unclassified;Geminiviridae;Begomovirus;Pepper yellow leaf curl Thailand virus    2742</p>
<p>...</p>
<p>After preparing the reference database and the list of viruses file, you can run FastViromeExplorer using the following command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>java -cp bin FastViromeExplorer -1 <span class="nv">$read1File</span> -2 <span class="nv">$read2File</span> -db /path-to-reference-database/<span class="nv">$reference</span>-database.fa -l /path-to-virus-list/<span class="nv">$virus</span>-list.txt -o <span class="nv">$outpurDirectory</span>
</pre></div>
</div>
</div>
<div class="section" id="usage">
<span id="usage"></span><h1>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h1>
<p>java -cp bin FastViromeExplorer -1 $read1File -2 $read2File -i $indexFile -o $outputDirectory</p>
<p>The full parameter list of FastViromeExplorer:</p>
<ol>
<li>-1: input .fastq file for read sequences (paired-end 1), mandatory field.</li>
<li>-2: input .fastq file for read sequences (paired-end 2).</li>
<li>-i: kallisto index file, mandatory field.</li>
<li>-db: reference database file in fasta/fa format.</li>
<li>-o: output directory, default option is the project directory.</li>
<li>-l: virus list containing all viruses present in the reference database along with their length.</li>
<li>-cr: the value of ratio criteria, default: 0.3.</li>
<li>-co: the value of coverage criteria, default: 0.1.</li>
<li>-cn: the value of number of reads criteria, default: 10.</li>
</ol><p>You can also run FastViromeExplorer with more flexible or more strict criteria by specifying different values for parameters &quot;-cr&quot;, &quot;-co&quot;, and &quot;-cn&quot;.</p>
</div>
<div class="section" id="support">
<span id="support"></span><h1>Support<a class="headerlink" href="#support" title="Permalink to this headline">¶</a></h1>
<p>If you are having issues, please contact us at saima5&#64;vt.edu</p>
</div>
<div class="section" id="license">
<span id="license"></span><h1>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h1>
<p>This project is licensed under the BSD 2-clause &quot;Simplified&quot; License.</p>
</div>
<div class="section" id="citation">
<span id="citation"></span><h1>Citation<a class="headerlink" href="#citation" title="Permalink to this headline">¶</a></h1>
<p>If you are using our tool, please cite us:</p>
<p>Saima Sultana Tithi, Roderick V. Jensen, and Liqing Zhang. &quot;FastViromeExplorer: A Pipeline for Virus and Phage Identification and Abundance Profiling in Metagenomics Data.&quot; bioRxiv (2017): 196998.</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="#">FastViromeExplorer</a></h1>








<h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2017, Saima Sultana Tithi.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/index.md.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>